core/src/main/java/org/elasticsearch/rest/action/admin/cluster/storedscripts/RestPutStoredScriptAction.java

```diff
@@ -0,0 +1,75 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.rest.action.admin.cluster.storedscripts;
+
+import org.elasticsearch.action.admin.cluster.storedscripts.PutStoredScriptRequest;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.support.AcknowledgedRestListener;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+import static org.elasticsearch.rest.RestRequest.Method.PUT;
+
+public class RestPutStoredScriptAction extends BaseRestHandler {
+
+    @Inject
+    public RestPutStoredScriptAction(Settings settings, RestController controller, Client client) {
+        this(settings, controller, true, client);
+    }
+
+    protected RestPutStoredScriptAction(Settings settings, RestController controller, boolean registerDefaultHandlers, Client client) {
+        super(settings, client);
+        if (registerDefaultHandlers) {
+            controller.registerHandler(POST, "/_scripts/{lang}/{id}", this);
+            controller.registerHandler(PUT, "/_scripts/{lang}/{id}", this);
+
+            controller.registerHandler(PUT, "/_scripts/{lang}/{id}/_create", new CreateHandler(settings, controller, client));
+            controller.registerHandler(POST, "/_scripts/{lang}/{id}/_create", new CreateHandler(settings, controller, client));
+        }
+    }
+
+    final class CreateHandler extends BaseRestHandler {
+        protected CreateHandler(Settings settings, RestController controller, Client client) {
+            super(settings, client);
+        }
+
+        @Override
+        public void handleRequest(RestRequest request, RestChannel channel, final Client client) {
+            request.params().put("op_type", "create");
+            RestPutStoredScriptAction.this.handleRequest(request, channel, client);
+        }
+    }
+
+    protected String getScriptLang(RestRequest request) {
+        return request.param("lang");
+    }
+
+    @Override
+    public void handleRequest(final RestRequest request, final RestChannel channel, Client client) {
+        PutStoredScriptRequest putRequest = new PutStoredScriptRequest(getScriptLang(request), request.param("id"));
+        putRequest.script(request.content());
+        client.admin().cluster().putStoredScript(putRequest, new AcknowledgedRestListener<>(channel));
```

Comment:
Should these Rest Actions not forward the request to the Transport Actions so that we make sure that are always handled in the same way? I thought in general the rest actions usually forward to the transport actions so the REST API and Java API are guaranteed to work the same?